docs/user-guide/private-repositories.md

```diff
@@ -41,24 +41,32 @@ The Argo CD UI don't support configuring SSH credentials. The SSH credentials ca
 argocd repo add git@github.com:argoproj/argocd-example-apps.git --ssh-private-key-path ~/.ssh/id_rsa
 ```
 
-## Self-Signed Certificates
+## HTTPS Self-signed Certificate
 
-If you are using self-hosted Git hosting service with the self-signed certificate then you need to disable certificate validation for that Git host.
-Following options are available:
+We do not currently have first-class support for this. See [#1513](https://github.com/argoproj/argo-cd/issues/1513).
+
+As a work-around, you can customize your Argo CD image. See [#1344](https://github.com/argoproj/argo-cd/issues/1344#issuecomment-479811810)
 
-Add repository using Argo CD CLI and `--insecure-ignore-host-key` flag:
+## Unknown SSH Hosts
 
+If you are using a self-hosted Git service over SSH then you need to disable host verification for that Git host.
+
+Following options are available:
+
+(1) Add repository using Argo CD CLI and `--insecure-ignore-host-key` flag:
```

Comment:
This should be option (2), since this is insecure.

Also, add a warning:
!!! warning "The `--insecure-ignore-host-key` should not be used in production as this is subject to man-in-the-middle attacks"